synthesizer/process/src/stack/call/dynamic.rs

```diff
@@ -588,19 +632,377 @@ impl<N: Network> CallTrait<N> for CallDynamic<N> {
+                                        input_output_index: (num_inputs + operand_index) as u16,
+                                    });
+                                },
+                                (
+                                    Value::DynamicRecord(record_dynamic),
+                                    OutputID::DynamicRecord(id_dynamic),
+                                    ValueType::DynamicRecord,
+                                    Value::Record(record_static),
+                                    OutputID::ExternalRecord(id_static),
+                                    ValueType::ExternalRecord(record_locator),
+                                ) => {
+                                    let program_id = *record_locator.program_id();
+                                    let record_name = *record_locator.resource();
+
+                                    let translation_proving_key =
+                                        get_record_translation_proving_key(&program_id, &record_name, rng)?;
+                                    
+                                    translation_data.push(RecordTranslationData {
+                                        // TODO: consider using a mapping from (program_id, record_name) to (proving_key, other data)
+                                        translation_proving_key,
+                                        record_static: record_static.clone(),
+                                        record_dynamic: record_dynamic.clone(),
+                                        program_id,
+                                        function_id: callee_console_function_id,
+                                        record_name,
+                                        is_input: false,
+                                        static_is_external: true,
+                                        tvk: *callee_request.tvk(),
+                                        record_view_key: None,
+                                        gamma: None,
+                                        id_static: *id_static,
+                                        id_dynamic: *id_dynamic,
+                                        input_output_index: (num_inputs + operand_index) as u16,
+                                    });
+                                },
+                                _ => {} // No translation to perform.
+                            }
+                        }
 
                         // Return the caller's request and response.
-                        (request_verification_inputs, caller_response_outputs)
+                        (callee_request_verification_inputs, caller_response.outputs().to_vec(), Some(translation_data))
                     }
                 }
             };
             lap!(timer, "Computed the request and response");
 
+            // TODO(dynamic_dispatch): If we let Registers keep e.g. an Arc<Stack>, we can just access Registers above.
```

Comment:
Is it possible to directly accumulate the `RecordTranslationData` in the `Trace`?